### PR TITLE
APPS-9923 - Do not add video codec if it already defined

### PR DESCRIPTION
**Pull request by e1adn**
APPS-9923 - Do not add video codec if it already defined

### CODE REVIEW

File: src/plugins/cloudinary/models/video-source/video-source.utils.js [hunk 1]
```diff
@@ -28,3 +30,31 @@ export function normalizeFormat(format) {
 
   return res;
 }
+
+const isContainCodec = (value) => Object.values(VIDEO_CODEC).some((codec) => value.includes(codec));
+
+export const isCodecAlreadyExist = (transformations, rawTransformation) => {
+  if (!transformations) {
+    return false;
```
Comment (maoznir) [inline src/plugins/cloudinary/models/video-source/video-source.utils.js:38]: what if the codec is in the raw transformation?

File: src/plugins/cloudinary/common.js [hunk 2]
```diff
@@ -178,17 +178,25 @@ const h264avcToString = (s) => {
...
-    case 'hev1':
-      return { video_codec: 'h265' };
-    case 'h264':
-      return { video_codec: 'h264:baseline:3.0' };
+    case VIDEO_CODEC.VP9:
+      return { video_codec: VIDEO_CODEC.VP9 };
+    case VIDEO_CODEC.HEV1:
+      return { video_codec: VIDEO_CODEC.H265 };
+    case VIDEO_CODEC.H264:
+      return { video_codec: `${VIDEO_CODEC}:baseline:3.0` };
```
Comment (maoznir) [inline src/plugins/cloudinary/common.js:199]: missing .H264 ? 
if it really is missing, then we're missing a test for this

Comment (maoznir) [reply src/plugins/cloudinary/common.js:199]: Also, this is very verbose and surly can be refactored to something like:
if (isCodec(...)){
  return {video_codec: codec}
}

File: src/plugins/cloudinary/models/video-source/video-source.utils.js [hunk 3]
```diff
@@ -28,3 +30,31 @@ export function normalizeFormat(format) {
 
   return res;
 }
+
+const isContainCodec = (value) => Object.values(VIDEO_CODEC).some((codec) => value.includes(codec));
```
Comment (maoznir) [inline src/plugins/cloudinary/models/video-source/video-source.utils.js:34]: please make sure there are passing unit tests for the utils

Comment (maoznir) [reply src/plugins/cloudinary/models/video-source/video-source.utils.js:34]: You should also add an integration test for the specific case this pr is handling

**Review by maoznir — CHANGES_REQUESTED** [review]

File: test/unit/videoSource.test.js [hunk 4]
```diff
@@ -297,3 +298,66 @@ describe('tests withCredentials', () => {
 });
 
 
+describe('test isCodecAlreadyExist method', () => {
```
Comment (maoznir) [inline test/unit/videoSource.test.js:301]: i'd also add tests for the generated url, to see that codec isn't added or is added when needed

**Review by maoznir — CHANGES_REQUESTED** [review]

**Review by maoznir — APPROVED** [review]

File: test/unit/videoSource.test.js (hunk 4 above)
Comment (e1adn) [reply test/unit/videoSource.test.js:301]: @maoznir done

**Pull request merged**